src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -63,11 +61,18 @@ public class PooledQueueCache<TQueueMessage, TCachedMessage>
         }
 
         /// <summary>
-        /// Called with the newest item in the cache and last item purged after a cache purge has run.
-        /// For ordered reliable queues we shouldn't need to notify on every purged event, only on the last event 
-        ///   of every set of events that get purged.
+        /// Cached message count
         /// </summary>
-        public Action<TCachedMessage?,TCachedMessage?> OnPurged { get; set; }
+        public int ItemCount { get
```

Comment:
This is relatively expensive, and doe not look necessary.  All we need to know is if the cache is empty, correct?